lib/util/createDomain.js

```diff
@@ -11,9 +11,13 @@ module.exports = function createDomain(options, listeningApp) {
   const hostname = options.useLocalIp ? internalIp.v4() : options.host;
 
   // the formatted domain (url without path) of the webpack server
-  return options.public ? `${protocol}://${options.public}` : url.format({
-    protocol,
-    hostname,
-    port
-  });
+  if (options.public) {
+    return options.public.indexOf('://') > 0 ? `${options.public}` : `${protocol}://${options.public}`;
```

Comment:
this is not a good way to check a url for a protocol. please use [`url.parse`](https://nodejs.org/api/url.html#url_url_parse_urlstring_parsequerystring_slashesdenotehost) instead, and examine the result for a protocol.